subprojects/test-runner/service/src/main/kotlin/com/avito/runner/service/worker/DeviceWorker.kt

```diff
@@ -83,7 +83,7 @@ internal class DeviceWorker(
                             result = result
                         )
                     }.onFailure { failure ->
-                        onDeviceDieOnIntention(intention, failure)
+                        onDeviceDieWhenActionExecuted(intention, failure)
```

Comment:
Before: If we have a bug on execution logic then the device will die and test re-executes. So it leads to infinite re-execution
Now: If we have a failure at the execution phase then we will kill a device and make test execution lost

Reply:
In the future, we should make device dying more accurate